lib/greenhouse_api/client.rb

```diff
@@ -1,169 +1,59 @@
-# typed: true
+# typed: false
 # frozen_string_literal: true
 
-require 'faraday'
-require 'base64'
-require 'json'
-require 'sorbet-runtime'
+require_relative 'base_client'
+require_relative 'resources/candidates'
+require_relative 'resources/offers'
+require_relative 'resources/users'
 
 module GreenhouseApi
-  class Response
-    attr_reader :headers, :body, :status
-
-    def initialize(headers:, body:, status:)
-      @headers = headers
-      @body = body
-      @status = status
-    end
-  end
-
   class Client
-    extend T::Sig
-
-    MAX_PER_PAGE = 500
-    ON_BEHALF_OF = 'On-Behalf-Of'
-    API_URL = 'https://harvest.greenhouse.io'
-
     def initialize(api_key)
       @api_key = api_key
     end
 
+    def list_many(resource, params = {})
+      base_client.list_many(resource, params)
+    end
+    
     def list_candidates(params = {})
```

Comment:
nice. so it looks like this isn't going to introduce any breaking changes, right?
i see that a consumer could alternatively instantiate a candidate client

Reply:
Yup, they can either instantiate a client or specific client (i.e. candidates), so no breaking changes